OccuPoint[] sensor_data
int32[] persist_vec
---
ProbPoint[] posterior
